java/arcs/core/allocator/Allocator.kt

```diff
@@ -116,12 +116,33 @@ class Allocator(val hostRegistry: HostRegistry) {
         storageKey: CreateableStorageKey,
         type: Type
     ): StorageKey =
-        CapabilitiesResolver(CapabilitiesResolver.StorageKeyOptions(arcId)).createStorageKey(
-            storageKey.capabilities, type.toSchemaHash()
-        )?.childKeyForHandle(idGenerator.newChildId(arcId, "").toString())
-            ?: throw Exception(
-                "Unable to create storage key $storageKey"
+        CapabilitiesResolver(CapabilitiesResolver.CapabilitiesResolverOptions(arcId))
+            .createStorageKey(
+                storageKey.capabilities,
+                toEntitySchema(type),
+                idGenerator.newChildId(arcId, "").toString()
```

Comment:
Is there any reason for us to pass in the unique parameter for debugging? Would putting the recipe handle name/id there be useful?

Reply:
the storageKey doesn't really care about the recipe name, so just a unique id should be fine, i think.
please, note that i moved calling the `childKeyForHandle` from outside into inside of CapabilitiesResolver (only in Kt, the ts change is coming soon) - we talked about it yesterday, and considering that CapabilitiesResolver only creates storage keys for handles atm, and that it accepts an entitySchema as parameter, it makes more sense, i think.